src/pages/Products/Products.tsx

```diff
@@ -76,21 +75,30 @@ const Products = () => {
         </div>
         <div className={cls.products_info}>
           <h2>Total Product</h2>
-          <div className={cls.count}>{totalProducts.length}</div>
+          <div className={cls.count}>
+            {productsStore.totalProductsList.length}
+          </div>
         </div>
       </div>
       <InfiniteScroll
         className={cls.infinite_scroll}
-        dataLength={products.length}
-        next={() => setPage((prevPage) => prevPage + 10)}
+        dataLength={productsStore.totalProductsList.length}
+        next={() => productsStore.getProductsList()}
```

Comment:
1. если сохранять передачу сюда `() => productsStore.getProductsList()`, то нужно вынести в колбэк и обернуть в useCallback
2. а вообще можно вот так просто сделать:
```suggestion
        next={productsStore.getProductsList}
```